ompi/mca/bcol/basesmuma/bcol_basesmuma_buf_mgmt.c

```diff
@@ -99,6 +99,9 @@ int bcol_basesmuma_free_buff( sm_buffer_mgmt * buff_block,
      *   associated with this bank have been freed.
      */
     assert(generation == buff_block->ctl_buffs_mgmt[memory_bank].bank_gen_counter);
+#if !OPAL_ENABLE_DEBUG
```

Comment:
Just curious -- is there a reason not to put the `assert()` inside `#if OPAL_ENABLE_DEBUG`?  That would obviate the need for the spurious `(void)generation` line.